lib/ConsumerImpl.cc

```diff
@@ -319,6 +320,27 @@ void ConsumerImpl::unsubscribeAsync(ResultCallback originalCallback) {
     }
 }
 
+void ConsumerImpl::triggerCheckExpiredChunkedTimer() {
+    checkExpiredChunkedTimer_ = executor_->createDeadlineTimer();
+    checkExpiredChunkedTimer_->expires_from_now(
+        boost::posix_time::milliseconds(expireTimeOfIncompleteChunkedMessageMs_));
+    checkExpiredChunkedTimer_->async_wait([this](const boost::system::error_code& ec) -> void {
+        if (ec) {
+            LOG_DEBUG(getName() << " Check expired chunked messages was failed or cancelled, code[" << ec
+                                << "].");
+            return;
+        }
+        Lock lock(chunkProcessMutex_);
+        long currentTimeMs = TimeUtils::currentTimeMillis();
+        chunkedMessageCache_.removeOldestValuesIf(
+            [this, &currentTimeMs](const std::string& uuid, const ChunkedMessageCtx& ctx) -> bool {
```

Comment:
Just capture `expireTimeOfIncompleteChunkedMessageMs_ ` on here.

Reply:
Are there any concerns here?

Reply:
Oh, You still caught `this` above.

Reply:
Are there any concerns here? I still don't understand what the benefit of this approach. I think may be the effect is the same?

Reply:
Yes, no problem here. I mainly want to express that this is captured here: https://github.com/apache/pulsar-client-cpp/pull/71#discussion_r1007591353

Now that you've explained it, you can resolve it.